### PR TITLE
Parse more fields from elasticsearch audit log

### PR DESCRIPTION
**Pull request by ycombinator**
Follow up to https://github.com/elastic/beats/pull/10352 per https://github.com/elastic/beats/pull/10352#issuecomment-457933083:

> While working on this PR I realized that we don't have sample lines for the **structured** elasticsearch audit log containing a request body (which is supposed to be parsed into the `http.request.body.content` field). I'm working with `@albertzaharovits` to get such a sample and will incorporate it into follow up PRs (for `master` and `6.x`).

Accordingly, this PR adds sample lines to the structured and unstructured log file test fixtures for the `elasticsearch/audit` fileset and teaches the fileset to parse any new fields encountered in these sample lines.

### CODE REVIEW

**Comment by ycombinator** [comment]
jenkins, test this

File: filebeat/module/elasticsearch/audit/_meta/fields.yml [hunk 1]
```diff
@@ -22,10 +26,18 @@
       description: "The name of the action that was executed"
       example: "cluster:monitor/main"
       type: keyword
+    - name: url.params.text
+      description: "REST URI parameters"
+      example: "{username=jacknich2}"
+      type: text
```
Comment (ruflin) [inline filebeat/module/elasticsearch/audit/_meta/fields.yml]: text? Was this intended to be a multifield?

File: filebeat/module/elasticsearch/audit/ingest/pipeline-plaintext.json [hunk 2]
```diff
@@ -11,17 +11,18 @@
                     "ES_AUDIT_EVENT_TYPE": "\\[%{WORD:event.type}\\]",
                     "ES_AUDIT_ORIGIN_TYPE": "(origin_type\\=\\[%{WORD:elasticsearch.audit.origin.type}\\])?",
                     "ES_AUDIT_ORIGIN_ADDRESS": "(origin_address\\=\\[%{IPORHOST:source.ip}\\])?",
-                    "ES_AUDIT_PRINCIPAL": "(principal\\=\\[%{WORD:user.name}\\])?",
-                    "ES_AUDIT_REALM": "(realm\\=\\[%{WORD:elasticsearch.audit.user.realm}\\])?",
+                    "ES_AUDIT_PRINCIPAL": "(principal\\=\\[%{DATA:user.name}\\])?",
+                    "ES_AUDIT_REALM": "(realm\\=\\[%{WORD:elasticsearch.audit.realm}\\])?",
```
Comment (ruflin) [inline filebeat/module/elasticsearch/audit/ingest/pipeline-plaintext.json]: Breaking change or was never release?

File: filebeat/module/elasticsearch/audit/_meta/fields.yml (hunk 1 above)
Comment (ycombinator) [reply filebeat/module/elasticsearch/audit/_meta/fields.yml]: Yeah, lets make this a multi-field. Changing...

File: filebeat/module/elasticsearch/audit/ingest/pipeline-plaintext.json (hunk 2 above)
Comment (ycombinator) [reply filebeat/module/elasticsearch/audit/ingest/pipeline-plaintext.json]: I think this qualifies as a breaking change. Previously we thought that the value of `realm=[...]` was the authenticated user's realm (`es.audit.user.realm`) but it turns out to be the realm that was used to test against before the user was authenticated or not (`es.audit.realm`).

The JSON audit logs contain both fields, but the plaintext audit logs only contain the latter one (`es.audit.realm`). Hence this change.

From a ES mapping perspective, we're introducing a new field here, `es.audit.realm` so in that sense it's not a breaking change.

**Pull request merged**